packages/arb-token-bridge-ui/src/hooks/useWithdrawals.ts

```diff
@@ -0,0 +1,81 @@
+import useSWRImmutable from 'swr/immutable'
+import {
+  fetchWithdrawals,
+  FetchWithdrawalsParams,
+  L2ToL1EventResultPlus
+} from 'token-bridge-sdk'
+import { PageParams } from '../components/TransactionHistory/TransactionsTable/TransactionsTable'
+import { useAppState } from '../state'
+import { MergedTransaction } from '../state/app/state'
+import { isPending, transformWithdrawals } from '../state/app/utils'
+import { useL2Gateways } from './useL2Gateways'
+import { useNetworksAndSigners } from './useNetworksAndSigners'
+
+export type CompleteWithdrawalData = {
+  withdrawals: L2ToL1EventResultPlus[]
+  pendingWithdrawals: L2ToL1EventResultPlus[]
+  transformedWithdrawals: MergedTransaction[]
+}
+
+const fetchCompleteWithdrawalData = async (
+  params: FetchWithdrawalsParams
+): Promise<CompleteWithdrawalData> => {
+  // get the original deposits
+  const withdrawals = await fetchWithdrawals(params)
+
+  // filter out pending withdrawals
+  const pendingWithdrawalMap = new Map<string, boolean>()
+  const completeWithdrawalData = transformWithdrawals(
+    withdrawals.sort((msgA, msgB) => +msgB.timestamp - +msgA.timestamp)
+  )
+
+  completeWithdrawalData.forEach(completeTxData => {
+    if (isPending(completeTxData)) {
+      pendingWithdrawalMap.set(completeTxData.txId, true)
+    }
+  })
+  const pendingWithdrawals = withdrawals.filter(
+    tx => tx.l2TxHash && pendingWithdrawalMap.get(tx.l2TxHash)
```

Comment:
Hm why do we check `tx.l2TxHash` here? Can it be not present? Also let's be more explicit that we're checking if an element belongs to the map